### PR TITLE
CLI now correctly parses duration

### PR DESCRIPTION
**Pull request by CluEleSsUK**
- fixed a bug where duration strings after the first entry would be ignored, e.g. 1d15m would encrypt for only 1d and ignore the 15m
- fixed a bug where integer overflow could occur when adding a date too far in the future

### CODE REVIEW

File: cmd/tle/commands/encrypt.go [hunk 1]
```diff
@@ -41,61 +42,135 @@ func Encrypt(flags Flags, dst io.Writer, src io.Reader, network *http.Network) e
...
+			return out, nil
+		}
 
-	pieces := strings.Split(duration, "d")
-	if len(pieces) == 2 {
-		days, err := strconv.Atoi(pieces[0])
-		if err != nil {
-			return time.Second, fmt.Errorf("parse day duration: %w", err)
+		ints, remainingInput, err := parsePrecedingInt(i)
+		if ints == 0 || err != nil {
```
Comment (AnomalRoil) [inline cmd/tle/commands/encrypt.go:109]: is "1y0m2w1d0h5m" really not valid?

File: cmd/tle/commands/commands_test.go [hunk 2]
```diff
@@ -15,15 +18,65 @@ func Test_ParseDuration(t *testing.T) {
...
+			duration: "1C",
+			date:     time.Now(),
+			err:      ErrInvalidDuration,
+		},
+		{
+			name:     "parseMissingMultiplier",
+			duration: "DM",
+			date:     time.Now(),
+			err:      ErrInvalidDurationMultiplier,
+		},
```
Comment (AnomalRoil) [inline cmd/tle/commands/commands_test.go:74]: Can we add a test with 0s?
And one with repeating chars?
`"1y4m2y2d1h5m15s"` and `"1y0M2d0m5s"` maybe.

**Review by AnomalRoil — CHANGES_REQUESTED** [review]

File: cmd/tle/commands/commands_test.go (hunk 2 above)
Comment (CluEleSsUK) [reply cmd/tle/commands/commands_test.go:74]: should repetition == last one wins, or add them together in your opinion? My initial feeling was the former

Comment (CluEleSsUK) [reply cmd/tle/commands/commands_test.go:74]: have pushed a change where last one wins

**Review by AnomalRoil — APPROVED** [review]
LGTM

**Pull request merged**